### ISSUE TITLE
[DRY] More redundant `zero` fields in `Algebra.Structures`

### ISSUE THREAD

**Issue by jamesmckinna** [agda/agda-stdlib#2396]
Cf. #2195 and #2253 

* `IsNonAssociativeRing` admits enough properties to make the `zero` field redundant by exactly the same argument as for `IsRingWithoutOne` (so the redundancy is a `bug`,  and removing  it would be `breaking`)
* correspondingly, the argument for `IsRingWithoutOne` makes no use of `*-assoc`...

Possible remedy: introduce a further refinement in which a new `IsNonAssociativeRingWithoutOne` would the 'right' home for such an argument, with each of the above two structures inheriting from that...?

**Comment by jamesmckinna** [comment]
@lexvanderstoep Are you planning a PR at any point in this direction?

**Comment by lexvanderstoep** [comment]
Yes @jamesmckinna, there’s https://github.com/agda/agda-stdlib/pull/2410 which was shelved at the time due to it being a breaking change.

What do you think? Is it a better time now?

**Comment by jamesmckinna** [comment]
@lexvanderstoep apologies! I'd completely forgotten (and had `draft:false` turned on while looking back over open PRs... d'oh!).
We'll be discussing a new release, and the development path after that, in early January, so let's review this then. In the meantime, can you see whether your PR builds against the current version of `master`? Thanks!

**Comment by lexvanderstoep** [comment]
Done! Still works like a charm.

I'll leave the PR as a draft for now then. We can come back to it when we get around to the new release :)

**Comment by jamesmckinna** [comment]
Closing, now that #2410 has finally landed!

We might always find more redundancies, but can start a fresh issue, as and when.
